src/zkevm_specs/evm/execution_result/call.py

```diff
@@ -0,0 +1,156 @@
+from ..opcode import Opcode
+from .execution_result import ExecutionResult
+
+
+def call(curr: Step, next: Step, r: int, opcode: Opcode):
+    # Verify opcode
+    assert opcode == Opcode.CALL
+
+    # Verify depth
+    depth = curr.call_context_lookup(CallContextTag.Depth)
+    curr.fixed_lookup(FixedTableTag.Range1024, [depth - 1])
+
+    # Gas needs full decompression due to EIP 150
+    bytes_gas = curr.decompress(curr.stack_pop_lookup(), 32, r)
+    bytes_callee_address = curr.decompress(curr.stack_pop_lookup(), 32, r)
+    bytes_value = curr.decompress(curr.stack_pop_lookup(), 32, r)
+    bytes_cd_offset = curr.decompress(curr.stack_pop_lookup(), 32, r)
+    bytes_cd_length = curr.decompress(curr.stack_pop_lookup(), 5, r)
+    bytes_rd_offset = curr.decompress(curr.stack_pop_lookup(), 32, r)
+    bytes_rd_length = curr.decompress(curr.stack_pop_lookup(), 5, r)
+    result = curr.stack_push_lookup()
+    assert_bool(result)
+
+    callee_address = le_to_int(bytes_callee_address[:20])
+    gas = le_to_int(bytes_gas[:8])
+
+    # Verify transfer
+    rw_counter_end_of_reversion = curr.call_context_lookup(CallContextTag.RWCounterEndOfReversion)
+    caller_address = curr.call_context_lookup(CallContextTag.CalleeAddress)
+    is_persistent = curr.call_context_lookup(CallContextTag.IsPersistent)
+    is_static = curr.call_context_lookup(CallContextTag.IsStatic)
+
+    has_value = not curr.is_zero(sum(bytes_value))
+    if has_value:
+        assert is_static == False
+    curr.assert_transfer(caller_address, callee_address, bytes_value,
+                         is_persistent, rw_counter_end_of_reversion, r)
+
+    # Verify memory expansion
+    next_memory_size, memory_gas_cost = curr.assert_memory_expansion(
+        bytes_cd_offset, bytes_cd_length, bytes_rd_offset, bytes_rd_length)
+
+    # Verify gas cost
+    tx_id = curr.call_context_lookup(CallContextTag.TxId)
+    is_cold_access = 1 - curr.w_lookup(RWTableTag.TxAccessListAccount, [tx_id, callee_address, 1],
+                                       is_persistent, rw_counter_end_of_reversion)[0]
+    code_hash = curr.r_lookup(RWTableTag.AccountCodeHash, [callee_address])[0]
+    is_empty_code_hash = curr.is_equal(code_hash, linear_combine(EMPTY_CODE_HASH, r))
+    callee_nonce = curr.r_lookup(RWTableTag.AccountNonce, [callee_address])[0]
+    callee_balance = curr.r_lookup(RWTableTag.AccountBalance, [callee_address])[0]
+    is_zero_nonce = curr.is_zero(callee_nonce)
+    is_zero_balance = curr.is_zero(callee_balance)
+    is_account_empty = is_zero_nonce and is_zero_balance and is_empty_code_hash
+    base_gas_cost = 100 \
+        + is_cold_access * 2500 \
+        + is_account_empty * 25000 \
+        + has_value * 9000 \
+        + memory_gas_cost
+
+    gas_available = curr.call.gas_left - base_gas_cost
+    one_64th_available_gas = le_to_int(curr.allocate_byte(8))
+    curr.fixed_lookup(FixedTableTag.Range64, [gas_available - 64 * one_64th_available_gas])
```

Comment:
need to assert the lookup ?

Reply:
The assertion is already done inside the function.